portfolio/src/main/webapp/script.js

```diff
@@ -56,50 +56,44 @@ function pauseProjectChange() {
   clearTimeout(changeCurrProj);
 }
 
-google.charts.load('current', {'packages':['corechart']});
+google.charts.load('current', {'packages':['timeline']});
 google.charts.setOnLoadCallback(drawChart);
 
-/** Creates a chart and adds it to the page. */
+/** 
+ * Fetches programming experience data and draws chart. 
+ * In the case of incorrectly formatted data, display error to user
+ */
 function drawChart() {
-  // Define data values for chart in format [Programming Language, Users]
-  var data = google.visualization.arrayToDataTable([
-    ['Programming Languages', 'Users'],
-    ['Java',  900],
-    ['JavaScript',  1000],
-    ['Python',  1170],
-    ['C/C++',  1250],
-    ['C#',  1530]
-  ]);
-
-  // Define title and view options for the chart
-  const options = {
-    title: 'Trending Programming Languages',
-    width: 500,
-    height: 500,
-    backgroundColor: '#111',
-    hAxis: {
-      textStyle: {
-        color: '#FFF'
-      }
-    },
-    vAxis: {
-      textStyle: {
-        color: '#ffffff'
-      }
-    },
-    legend: {
-      textStyle: {
-        color: '#ffffff'
-      }
-    },
-    titleTextStyle: {
-      color: '#ffffff'
+  fetch('/programming-data').then(function(response) {
+    const chartContainer = document.getElementById('chart-container');
+    if (response.status === 500) {
+      chartContainer.appendChild(createParagraphElement("Error parsing chart data"));
```

Comment:
The date parsing error is only one of a variety of reasons why the server could return a 500. Could the server include the error message in its response? Then the client could use that message when provided and use something more generic like "Error fetching chart data" when not provided.